### PR TITLE
Improved validation suite

### PR DESCRIPTION
**Pull request by mverzett**
### Added script to analyse FWK report and get timing plot

So far it looks like this:

image

The picture changes significantly between 100 and 1000 events. This looks reasonable given that the ID will take some time to execute.

For the record, for 100 events it looks like this (I checked that the inputs agree with the plot):

image

### minor bugfix

I found a small bug in how the input files were handled by the `VarParser`, now it is fixed

### More validation

Added to the validation:
   * a script that in principle should test and validate a PR automatically, use with caution, maybe in a different dev area. Should it remove the branch once done?
   * included GEN timing and size info
   * on top of printing out to screen the validation now produces a log file that can be embedded in a web page produced by my script [web.py](https://github.com/mverzett/.bin/blob/master/web.py) (I am a lousy web dev, this is as far as I could go)

### CODE REVIEW

File: BParkingNano/test/validate_pr.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#! /bin/env bash
+
+set -o errexit
+set -o nounset
+
+: ${CMSSW_BASE:?"CMSSW_BASE is not set!  Run cmsenv!"}
+
+PRID=$1
+
+git pull origin master
```
Comment (sarafiorendi) [inline BParkingNano/test/validate_pr.sh]: maybe add the option to specify the remote label, if someone does not call it "origin"?

Comment (mverzett) [reply BParkingNano/test/validate_pr.sh]: OK, done

File: BParkingNano/test/validate_pr.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+else
+    remote=origin
+fi
+
+set -o errexit
+set -o nounset
+
+: ${CMSSW_BASE:?"CMSSW_BASE is not set!  Run cmsenv!"}
+
+git pull $remote master
```
Comment (sarafiorendi) [inline BParkingNano/test/validate_pr.sh]: I would replace this line with
git fetch $remote

File: BParkingNano/test/validate_pr.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+git checkout $remote/master -b official_current_master
+cd $CMSSW_BASE/src
+scram b -j 4
+cd $CMSSW_BASE/src/PhysicsTools/BParkingNano/test
+TAG=HEAD
+cmsRun run_nano_cfg.py reportEvery=10 tag=$TAG &> nano_$TAG'_data.log'
+cmsRun run_nano_cfg.py reportEvery=10 tag=$TAG isMC=True &> nano_$TAG'_mc.log'
+
+git fetch origin pull/$PRID/head:TEST_PR$PRID
```
Comment (sarafiorendi) [inline BParkingNano/test/validate_pr.sh]: origin -> $remote

**Pull request merged**